xcarchive/xcarchive.go

```diff
@@ -65,3 +67,26 @@ func GetEmbeddedInfoPlistPath(xcarchivePth string) (string, error) {
 func getAppSubfolder(basepth string) string {
 	return filepath.Join(basepth, "Products", "Applications")
 }
+
+func findDSYMs(archivePath string) ([]string, []string, error) {
+	dsymsDirPth := filepath.Join(archivePath, "dSYMs")
+	dsyms, err := utility.ListEntries(dsymsDirPth, utility.ExtensionFilter(".dsym", true))
+	if err != nil {
+		return []string{}, []string{}, err
+	}
+
+	appDSYMs := []string{}
+	frameworkDSYMs := []string{}
+	for _, dsym := range dsyms {
+		if strings.HasSuffix(dsym, ".app.dSYM") {
+			appDSYMs = append(appDSYMs, dsym)
+		} else {
+			frameworkDSYMs = append(frameworkDSYMs, dsym)
+		}
+	}
+	if len(appDSYMs) == 0 && len(frameworkDSYMs) == 0 {
+		return []string{}, []string{}, fmt.Errorf("no dsym found")
```

Comment:
Could you please also add a test for this error case?

Reply:
Sure thing, matey.